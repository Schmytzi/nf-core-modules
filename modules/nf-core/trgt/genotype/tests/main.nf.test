// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test trgt
nextflow_process {

    name "Test Process TRGT"
    script "../main.nf"
    process "TRGT"

    tag "modules"
    tag "modules_nfcore"
    tag "trgt"
    tag "trgt/genotype"

    setup {
        run("GUNZIP"){
            script "../../../gunzip/main.nf"
            process {
                """
                input[0] = [
                    [ id : 'chr22' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr22_chr22_KI270734v1_random/sequence/genome.fa.gz', checkIfExists: true)
                ]
            """
            }
        }
        run("SAMTOOLS_FAIDX"){
            script "../../../samtools/faidx/main.nf"
            process {
                """
                input[0] = GUNZIP.out.gunzip
                input[1] = [[],[]]
                """
            }
        }
    }

    test("homo sapiens - [bam,bai,XX], [fa,fai], bed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam.bai', checkIfExists: true),
                    'XX'
                ]
                input[1] = GUNZIP.out.gunzip
                input[2] = SAMTOOLS_FAIDX.out.fai
                input[3] = Channel.of('chr22\t18890357\t18890451\tID=TEST;MOTIFS=AT;STRUC=(AT)n')
                                .collectFile(name : 'repeats.bed', newLine: false)
                                .map { file -> [ [ id : 'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo sapiens - [bam,bai,[]], [fa,fai], bed") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam.bai', checkIfExists: true),
                    []
                ]
                input[1] = GUNZIP.out.gunzip
                input[2] = SAMTOOLS_FAIDX.out.fai
                input[3] = Channel.of('chr22\t18890357\t18890451\tID=TEST;MOTIFS=AT;STRUC=(AT)n')
                                .collectFile(name : 'repeats.bed', newLine: false)
                                .map { file -> [ [ id : 'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo sapiens - [bam,bai,XX], [fa,fai], bed - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam.bai', checkIfExists: true),
                    'XX'
                ]
                input[1] = GUNZIP.out.gunzip
                input[2] = SAMTOOLS_FAIDX.out.fai
                input[3] = Channel.of('chr22\t18890357\t18890451\tID=TEST;MOTIFS=AT;STRUC=(AT)n')
                                .collectFile(name : 'repeats.bed', newLine: false)
                                .map { file -> [ [ id : 'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo sapiens - [bam,bai,[]], [fa,fai], bed - stub") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/pacbio/bam/test.sorted.bam.bai', checkIfExists: true),
                    []
                ]
                input[1] = GUNZIP.out.gunzip
                input[2] = SAMTOOLS_FAIDX.out.fai
                input[3] = Channel.of('chr22\t18890357\t18890451\tID=TEST;MOTIFS=AT;STRUC=(AT)n')
                                .collectFile(name : 'repeats.bed', newLine: false)
                                .map { file -> [ [ id : 'chr22' ], file ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
