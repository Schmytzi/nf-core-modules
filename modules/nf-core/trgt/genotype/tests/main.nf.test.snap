{
    "homo sapiens - [bam,bai,XX], [fa,fai], bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,4e5b6028e3ea3d06829ca18af1cc492e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,645a2c199d04aaf733ce84247f0b0f37"
                    ]
                ],
                "2": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,645a2c199d04aaf733ce84247f0b0f37"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,4e5b6028e3ea3d06829ca18af1cc492e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-14T11:54:30.132679743"
    },
    "homo sapiens - [bam,bai,[]], [fa,fai], bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,0337e3287dcc71728c36c185dae7738f"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,8d6d1bf813109870692d2f20d59399e7"
                    ]
                ],
                "2": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,8d6d1bf813109870692d2f20d59399e7"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,0337e3287dcc71728c36c185dae7738f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-14T13:07:06.490237057"
    },
    "homo sapiens - [bam,bai,XX], [fa,fai], bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "/home/xschmy/projects/nf-core-modules/.nf-test/tests/d7351355c32211a2bffe015986fa0ad3/work/3a/4e872d9aa4463312c0c31d799b33a6/test.vcf.gz"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "/home/xschmy/projects/nf-core-modules/.nf-test/tests/d7351355c32211a2bffe015986fa0ad3/work/3a/4e872d9aa4463312c0c31d799b33a6/test.vcf.gz"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-14T13:05:49.03433982"
    },
    "homo sapiens - [bam,bai,[]], [fa,fai], bed": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,0337e3287dcc71728c36c185dae7738f"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,8d6d1bf813109870692d2f20d59399e7"
                    ]
                ],
                "2": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.spanning.bam:md5,8d6d1bf813109870692d2f20d59399e7"
                    ]
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.vcf.gz:md5,0337e3287dcc71728c36c185dae7738f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bbf36cf2b82bf043a106de7a7a36381d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-11-14T13:02:11.201480175"
    }
}